### PR TITLE
cmake: support components in find modules

### PR DESCRIPTION
**Pull request by abouvier**
It's more meaningful and it allows early rejection of `cpp-httplib` library with no OpenSSL support. ~~Update of `cpp-httplib` needed for https://github.com/yhirose/cpp-httplib/pull/1504.~~
Probably solve https://github.com/yuzu-emu/yuzu/issues/9780 too, thanks to a ugly hack.

### CODE REVIEW

File: CMakeModules/FindLLVM.cmake [hunk 1]
```diff
@@ -2,15 +2,24 @@
...
-if (LLVM_FOUND AND NOT TARGET LLVM::Demangle)
+if (LLVM_Demangle_FOUND AND NOT TARGET LLVM::Demangle)
     add_library(LLVM::Demangle INTERFACE IMPORTED)
-    llvm_map_components_to_libnames(LLVM_LIBRARIES demangle)
     target_compile_definitions(LLVM::Demangle INTERFACE ${LLVM_DEFINITIONS})
     target_include_directories(LLVM::Demangle INTERFACE ${LLVM_INCLUDE_DIRS})
+    # ugly hack because llvm_config doesn't support interface library
+    add_library(_dummy_lib SHARED EXCLUDE_FROM_ALL src/yuzu/main.cpp)
+    llvm_config(_dummy_lib USE_SHARED demangle)
+    get_target_property(LLVM_LIBRARIES _dummy_lib LINK_LIBRARIES)
```
[the remark this answers is not in the document]
Comment (abouvier) [reply CMakeModules/FindLLVM.cmake:23]: `llvm_map_components_to_libnames` only resolves to static libraries (i.e. `libLLVMDemangle.a`). `llvm_config` with `USE_SHARED` resolves to the shared `libLLVM.so` library instead. But if `LLVMDemangle` is not part of the shared library (as seems to be the case on openSUSE) this macro still resolves to the static library.

[the remark this answers is not in the document]
Comment (Tachi107) [reply CMakeModules/FindLLVM.cmake:23]: Makes sense. Could you please also add a reference to <https://github.com/llvm/llvm-project/issues/34593>, or something similar?

**Comment by liamwhite** [comment]
Conflicts after recent merge

**Comment by liamwhite** [comment]
Conflicts again

File: CMakeModules/Findhttplib.cmake [hunk 2]
```diff
@@ -6,13 +6,23 @@ include(FindPackageHandleStandardArgs)
...
     pkg_search_module(HTTPLIB QUIET IMPORTED_TARGET cpp-httplib)
+    if ("-DCPPHTTPLIB_OPENSSL_SUPPORT" IN_LIST HTTPLIB_CFLAGS_OTHER)
+        set(httplib_OpenSSL_FOUND TRUE)
+    endif()
+    if ("-DCPPHTTPLIB_ZLIB_SUPPORT" IN_LIST HTTPLIB_CFLAGS_OTHER)
+        set(httplib_ZLIB_FOUND TRUE)
+    endif()
+    if ("-DCPPHTTPLIB_BROTLI_SUPPORT" IN_LIST HTTPLIB_CFLAGS_OTHER)
+        set(httplib_Brotli_FOUND TRUE)
+    endif()
```
Comment (liamwhite) [inline CMakeModules/Findhttplib.cmake:21]: Are we testing for these elsewhere?

Comment (abouvier) [reply CMakeModules/Findhttplib.cmake:21]: I don't think so, it was completely gratuitous!

**Review by liamwhite — APPROVED** [review]

**Pull request merged**